stdlib/TOML/test/values.jl

```diff
@@ -172,6 +172,6 @@ end
 @testset "Array" begin
     @test testval("[1,2,3]", Int64[1,2,3])
     @test testval("[1.0, 2.0, 3.0]", Float64[1.0, 2.0, 3.0])
-    @test testval("[1.0, 2.0, 3]", Union{Int64, Float64}[1.0, 2.0, Int64(3)])
```

Comment:
Just os I know, what are the rules know when it will promote to `Any`? IIRC, it used to keep it as a union up to two different eltypes but that has changed now?

Reply:
This now specializes for `Vector{String}`, `Vector{Bool}`, `Vector{(U)Int}`, and `Vector{Float64}`

Anything else is returned as `Vector{Any}`